### ISSUE TITLE
Redis "LOADING Redis is loading the dataset" error not handled

### ISSUE THREAD

**Issue by brunoqc** [slact/nchan#193]
Every time I reboot my server for a kernel update, nchan stops working.

It seems that the problem is that redis doesn't start before nginx.

I will add redis-server as a "Required-Start" thing in the init.d script but I was wondering if it would be good or not for nchan to handle this.

```
2016/06/10 12:09:31 [error] 2628#2628: Nchan: Can't connect to redis at 127.0.0.1:6379: Connection refused
2016/06/10 12:09:31 [error] 2627#2627: Nchan: Can't connect to redis at 127.0.0.1:6379: Connection refused
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script add_fakesub :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script channel_keepalive :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script delete :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script find_channel :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script get_message :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script get_message_from_key :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script publish :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script publish_status :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script rsck :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script subscriber_register :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: nchan: Failed loading redis lua script subscriber_unregister :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:33 [error] 2628#2628: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2628#2628: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script add_fakesub :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script channel_keepalive :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script delete :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script find_channel :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script get_message :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script get_message_from_key :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script publish :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script publish_status :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script rsck :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script subscriber_register :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: nchan: Failed loading redis lua script subscriber_unregister :LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:34 [error] 2627#2627: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:38 [error] 2627#2627: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:38 [error] 2627#2627: REDIS_REPLY_ERROR: LOADING Redis is loading the dataset in memory
2016/06/10 12:09:39 [error] 2627#2627: REDIS_REPLY_ERROR: NOSCRIPT No matching script. Please use EVAL.
2016/06/10 12:09:39 [error] 2627#2627: REDIS_REPLY_ERROR: NOSCRIPT No matching script. Please use EVAL.
2016/06/10 12:09:40 [error] 2627#2627: REDIS_REPLY_ERROR: NOSCRIPT No matching script. Please use EVAL.
[...] the last line repeats for ever
```

**Comment by slact** [comment]
Ah, right, I don't handle the "redis is still loading the dataset" condition yet. Nchan should definitely do this, because Redis can take a few minutes to load up with large datasets.

**Comment by slact** [comment]
fixed in 8555969

**Comment by brunoqc** [comment]
Seems fine. Thanks!
